File: sdk/go/invitations.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
	"strconv"
)

type Invitation struct {
	ID          string         `json:"id"`
	TenantID    string         `json:"tenant_id"`
	Email       string         `json:"email"`
	Role        string         `json:"role,omitempty"`
	Status      string         `json:"status"`
	InvitedBy   string         `json:"invited_by,omitempty"`
	ExpiresAt   string         `json:"expires_at,omitempty"`
	Metadata    map[string]any `json:"metadata,omitempty"`
	CreatedAt   string         `json:"created_at"`
	UpdatedAt   string         `json:"updated_at,omitempty"`
}

type CreateInvitationInput struct {
	Email         string         `json:"email"`
	TenantID      string         `json:"tenant_id"`
	Role          string         `json:"role,omitempty"`
	ExpiresInDays int            `json:"expires_in_days,omitempty"`
	Metadata      map[string]any `json:"metadata,omitempty"`
}

type InvitationPage struct {
	Data       []Invitation
	NextCursor string
}

type Invitations struct{ http *httpClient }

func (r *Invitations) Create(ctx context.Context, in CreateInvitationInput) (*Invitation, error) {
	var out Invitation
	err := r.http.do(ctx, http.MethodPost, "/v1/invites", nil, in, &out, false)
	return &out, err
}

func (r *Invitations) Get(ctx context.Context, id string) (*Invitation, error) {
	var out Invitation
	err := r.http.do(ctx, http.MethodGet, "/v1/invites/"+url.PathEscape(id), nil, nil, &out, true)
	return &out, err
}

func (r *Invitations) Delete(ctx context.Context, id string) error {
	return r.http.do(ctx, http.MethodDelete, "/v1/invites/"+url.PathEscape(id), nil, nil, nil, true)
}

func (r *Invitations) Resend(ctx context.Context, id string) error {
	return r.http.do(ctx, http.MethodPost, "/v1/invites/"+url.PathEscape(id)+"/resend", nil, struct{}{}, nil, false)
}

func (r *Invitations) List(ctx context.Context, params ListParams) (*InvitationPage, error) {
	q := url.Values{}
	if params.Tenant != "" {
		q.Set("tenant", params.Tenant)
	}
	if params.Limit > 0 {
		q.Set("limit", strconv.Itoa(params.Limit))
	}
	if params.Cursor != "" {
		q.Set("cursor", params.Cursor)
	}
	var env struct {
		Items      []Invitation `json:"items"`
		Data       []Invitation `json:"data"`
		NextCursor string       `json:"next_cursor"`
	}
	if err := r.http.do(ctx, http.MethodGet, "/v1/invites", q, nil, &env, true); err != nil {
		return nil, err
	}
	items := env.Items
	if items == nil {
		items = env.Data
	}
	return &InvitationPage{Data: items, NextCursor: env.NextCursor}, nil
}

File: sdk/js/node/src/invitations.ts
import type { HttpClient } from "./client.js";
import type { ListParams, Page } from "./users.js";

export interface Invitation {
  id: string;
  tenant_id: string;
  email: string;
  role?: string;
  status: string;
  invited_by?: string;
  expires_at?: string;
  metadata?: Record<string, unknown>;
  created_at: string;
  updated_at?: string;
}

export interface CreateInvitationInput {
  email: string;
  tenant_id: string;
  role?: string;
  expires_in_days?: number;
  metadata?: Record<string, unknown>;
}

interface ListEnvelope<T> {
  items?: T[];
  data?: T[];
  next_cursor?: string;
}

export class Invitations {
  constructor(private readonly http: HttpClient) {}

  create(input: CreateInvitationInput): Promise<Invitation> {
    return this.http.post<Invitation>("/v1/invites", input);
  }

  get(id: string): Promise<Invitation> {
    return this.http.get<Invitation>(`/v1/invites/${encodeURIComponent(id)}`);
  }

  delete(id: string): Promise<void> {
    return this.http.delete<void>(`/v1/invites/${encodeURIComponent(id)}`);
  }

  resend(id: string): Promise<void> {
    return this.http.post<void>(`/v1/invites/${encodeURIComponent(id)}/resend`, {});
  }

  async list(params: ListParams = {}): Promise<Page<Invitation>> {
    const res = await this.http.get<ListEnvelope<Invitation>>("/v1/invites", {
      query: { tenant: params.tenant, limit: params.limit, cursor: params.cursor },
    });
    return { data: res.items ?? res.data ?? [], nextCursor: res.next_cursor };
  }

  async *listAll(params: ListParams = {}): AsyncGenerator<Invitation> {
    let cursor = params.cursor;
    do {
      const page = await this.list({ ...params, cursor });
      for (const item of page.data) yield item;
      cursor = page.nextCursor;
    } while (cursor);
  }
}
